layer/flavors/catboost.py

```diff
@@ -20,7 +22,15 @@ def save_model_to_directory(
 
         mlflow.catboost.save_model(model_object, path=directory.as_posix())
 
-    def load_model_from_directory(self, directory: Path) -> ModelArtifact:
+    def load_model_from_directory(self, directory: Path) -> ModelRuntimeObjects:
         import mlflow.catboost
 
-        return mlflow.catboost.load_model(directory.as_uri())
+        model = mlflow.catboost.load_model(directory.as_uri())
+        return ModelRuntimeObjects(
+            model, lambda input_df: self.__predict(model, input_df)
+        )
+
+    @staticmethod
+    def __predict(model: Any, input_df: pd.DataFrame) -> pd.DataFrame:
```

Comment:
why do we have underscores we do expect our plugins to call this method.

Reply:
The plugin will call Model.predict(), i.e the object we return from layer.get_model(), so it's not going to be this one.

Reply:
I'm talking about `layer/contracts/models.py.`predict() in this PR.

Reply:
I see.  Further comments below:
- We can also change model parameter type to `ModelObject `
- Do we need this method to be static ?

Reply:
- Updated the code based on your first point.
- About the second, I think it should remain a static method. I don't see why it should be reading/writing state from the model flavor class instance. The other option is changing the model runtime object that has been loaded with mlflow. I don't think we should do that either.